src/axom/quest/ScatteredInterpolation.hpp

```diff
@@ -0,0 +1,402 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_SCATTERED_INTERPOLATION_H_
+#define QUEST_SCATTERED_INTERPOLATION_H_
+
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/sidre.hpp"
+#include "axom/primal.hpp"
+
+#include "axom/fmt.hpp"
+
+#include "conduit.hpp"
+#include "conduit_blueprint.hpp"
+
+#include <cmath>
+#include <limits>
+
+namespace
+{
+/// Helper function to extract the dimension from the coordinate values group
+/// of a mesh blueprint coordset
+inline int extractDimension(const conduit::Node& values_node)
+{
+  SLIC_ASSERT(values_node.has_child("x"));
+  return values_node.has_child("z") ? 3 : (values_node.has_child("y") ? 2 : 1);
+}
+
+/// Helper function to extract the number of points from the coordinate values group
+/// of a mesh blueprint coordset
+inline int extractSize(const conduit::Node& values_node)
+{
+  SLIC_ASSERT(values_node.has_child("x"));
+  return values_node["x"].dtype().number_of_elements();
+}
+
+/**
+ * \brief Utility function to create an axom::ArrayView over the array
+ * of native types stored by a conduit::Node
+ */
+template <typename T>
+inline axom::ArrayView<T> ArrayView_from_Node(conduit::Node& node, int sz)
+{
+  T* ptr = node.value();
+  return axom::ArrayView<T>(ptr, sz);
+}
+
+/**
+ * \brief Template specialization of ArrayView_from_Node for Point<double,2>
+ *
+ * \warning Assumes the underlying data is an MCArray with stride 2 access
+ */
+template <>
+inline axom::ArrayView<axom::primal::Point<double, 2>> ArrayView_from_Node(
+  conduit::Node& node,
+  int sz)
+{
+  using PointType = axom::primal::Point<double, 2>;
+
+  PointType* ptr = static_cast<PointType*>(node.data_ptr());
+  return axom::ArrayView<PointType>(ptr, sz);
+}
+
+/**
+ * \brief Template specialization of ArrayView_from_Node for Point<double,3>
+ *
+ * \warning Assumes the underlying data is an MCArray with stride 3 access
+ */
+template <>
+inline axom::ArrayView<axom::primal::Point<double, 3>> ArrayView_from_Node(
+  conduit::Node& node,
+  int sz)
+{
+  using PointType = axom::primal::Point<double, 3>;
+
+  PointType* ptr = static_cast<PointType*>(node.data_ptr());
+  return axom::ArrayView<PointType>(ptr, sz);
+}
+
+}  // namespace
+
+namespace axom
+{
+namespace quest
+{
+namespace detail
+{
+/**
+ * \brief Utility class to enable processing an array of points whose layout
+ * is either interleaved or separated strided arrays
+ */
+template <int NDIMS>
+struct InterleavedOrStridedPoints
```

Comment:
Note: This class is a kludge to allow processing of point arrays whose layout is either strided by coordinate, or whose coordinates are interleaved. Axom should provide a better native solution for this common situation.

I'll create an issue to track this.